salt/pillar/__init__.py

```diff
@@ -362,20 +362,24 @@ def get_tops(self):
         # Gather initial top files
         try:
             if self.opts['pillarenv']:
-                tops[self.opts['pillarenv']] = [
-                        compile_template(
-                            self.client.cache_file(
-                                self.opts['state_top'],
-                                self.opts['pillarenv']
-                                ),
-                            self.rend,
-                            self.opts['renderer'],
-                            self.opts['renderer_blacklist'],
-                            self.opts['renderer_whitelist'],
-                            self.opts['pillarenv'],
-                            _pillar_rend=True,
-                            )
-                        ]
+                # If file roots environment for the pillar does not match pillarenv, dont look for top file
+                if self.opts['pillarenv'] not in self.opts['file_roots']:
+                    log.debug("Pillarenv NOT equal to file_roots")
```

Comment:
This log message is inaccurate, it would never be equal to file_roots. Also, in Pillar compilation, we set file_roots equal to pillar_roots so that the same code that compiles states can compile Pillar SLS.

This should be changed to:

```python
log.debug('pillarenv (%s) not present in pillar_roots', self.opts['pillarenv'])
```